mythril/laser/ethereum/transaction/concolic.py

```diff
@@ -11,12 +11,13 @@ def execute_message_call(laser_evm, callee_address, caller_address, origin_addre
     del laser_evm.open_states[:]
 
     for open_world_state in open_states:
+        next_transaction_id = get_next_transaction_id()
         transaction = MessageCallTransaction(
-            identifier=get_next_transaction_id(),
+            identifier=next_transaction_id,
             world_state=open_world_state,
             callee_account=open_world_state[callee_address],
             caller=caller_address,
-            call_data=data,
+            call_data=Calldata(next_transaction_id, data),
```

Comment:
Shouldn't there be some set_global_state() happening somewhere?

Reply:
It's happening. In the `initial_global_state` function of MessageCallTransaction.